content/francineblanc/how-to-use-git-for-beginners.md

```diff
@@ -0,0 +1,278 @@
+``` 
+
+The `<filename>` is the name of the file to be unstaged.
+
+## Committing changes
+
+Committing is generally the last step of the Git workflow and is generally thought of as being a snapshot of a project at a particular time. In this stage, changes in the staging area are saved inside the local repo, which is achieved with the command `git commit -m` followed by a space and a short message explaining the commit in quotes. For example:
+
+```bash
+git commit -m "Add title and description to intro file"
+```
+
+There are general [conventions] for writing a commit message including: 
+- Making sure they are no more than 50 characters long.
+- Writing them in the imperative or present tense. 
+
+Messages should always be clear and informative to others who may go back to confirm what changes were made.
+
+Committing is an important step in the Git workflow, as once a change has been committed, it can be recalled at a later date or the project can be rewound to that particular version. Commit history can be viewed with `git log`, which will display a list of commits in chronological order (the most recent commits being at the top), along with information such as the author of the commit, the date and time of the commit and the commit message.
+
+### Undoing a commit
+
+In the situation where the last commit needs to be undone, the `git reset` command can again be used, with some modifications. 
+
+The current commit is called the `HEAD` commit, which is generally the most recently made commit. To find out which commit this is, the command `git show HEAD` can be executed in the terminal. This will display information about the most recent commit, including a unique 40 character SHA hash, which Git uses like an id to identify revisions in the repo.
+
+To undo the immediately previous commit, the command `git reset --soft HEAD~1` can be run, which will rewind the current `HEAD` commit to the immediately previous commit. The `--soft` flag ensures that any changes made to the files are preserved.
+
+If the rewind needs to go beyond the most recent commit, the command `git reset SHA`, where `SHA` is the first 7 characters of the SHA of the commit, can be used. Note that the SHA of all previous commits can be viewed using `git log`.
+## Git branching
+
+Previously, the output from running `git status` referred to being on `branch main`. Branches are a core feature in Git. They allow for new updates and features to be developed and tested without disrupting the `main` branch of the project.
+
+The first branch to be aware of is the `HEAD` branch, which is the currently active branch. So, the `main` branch referenced above was the currently active branch and in GitHub is the name of the default branch. In Git, the default branch name is `master`, which was also originally the case in GitHub however this was renamed for the reasons outlined [here]. The main take away though is that both `master` and `main` are used as default branch names.
```

Comment:
```suggestion
The `main` branch usually refers to the default branch that serves as the base of a project. In Git, the default branch name is `master`, which was also originally the case in GitHub. However this was renamed in GitHub for the reasons outlined [here]. The main take away though is that both `master` and `main` are used as default branch names.
```

I would change the link for `[here]` to point to this `README.md` about the change from `master` to `main` (as told by GitHub themselves): https://github.com/github/renaming/blob/main/README.md